docs/proposals/import-cluster.md

```diff
@@ -0,0 +1,56 @@
+# Adding an external cluster to Kubermatic
+
+**Author**: Lukasz Zajaczkowski (@zreigz)
+
+**Status**: Draft proposal; prototype in progress.
+
+## Goals
+The end user would like to add existing Kubernetes cluster to Kubermatic platform. Therefore, Kubermatic does not provision
+Kubernetes, but only sets up the connection to communicate with the cluster and install controllers to enable some Kubermatic features.
+Kubermatic features, including displaying cluster details, metrics and role-based access control, will be available for external clusters.
```

Comment:
Would we install the metrics-server ourselves, if it's not there?

Reply:
It would be good to have this for the metrics. It's correlated with your second question about kubeconfig privileges.

Reply:
Yes it would be good for metrics. But what if the metrics-server is not there? Will we install it or will we simply display a warning "No metrics available, please do X, Y Z to fix."?

Reply:
I have no clear answer for that. I think we should discuss some subjects during tech lead meeting

Reply:
We decided on not installing anything in the cluster ourselves, but rather document the requirements that must be met in order to external clusters to be shown our dashboard with all details and stuff.